RecoCTPPS/CTPPSPixelLocal/src/RPixDetClusterizer.cc

```diff
@@ -0,0 +1,179 @@
+  calib_rpix_digi_set_.clear();
+  for( std::set<CTPPSPixelDigi>::iterator RPdit = rpix_digi_set_.begin(); RPdit != rpix_digi_set_.end();++RPdit){
+
+    if((*RPdit).row() > maxRow || (*RPdit).column() > maxCol)
+       throw cms::Exception("CTPPSDigiOutofRange") 
+	 << " row = " << (*RPdit).row() << "  column = "<< (*RPdit).column();
+
+    unsigned char row = (*RPdit).row();
+    unsigned char column = (*RPdit).column();
+    unsigned short adc = (*RPdit).adc();
+    unsigned short electrons = 0;
+    std::pair<unsigned char, unsigned char> pixel = std::make_pair(row,column);
+
+// check if pixel is noisy or dead (i.e. in the mask)
+    const bool is_in = maskedPixels.find(pixel) != maskedPixels.end();
+    if(!is_in){
+//calibrate digi and store the new ones
+    electrons = calibrate(detId,adc,row,column,pcalibrations);
+    RPixCalibDigi calibDigi(row,column,adc,electrons);
+    calib_rpix_digi_set_.insert(calibDigi);
+    }
+  }
+  if(verbosity_) edm::LogInfo("RPixDetClusterizer")<<" RPix set size = "<<calib_rpix_digi_set_.size();
+
+// storing the seeds
+  SeedVector_.clear();
+  for (const auto &rpcd : calib_rpix_digi_set_){
+    if(rpcd.electrons() > SeedADCThreshold_*ElectronADCGain_){
+      SeedVector_.push_back(rpcd);
+    }
+  }
+  if(verbosity_) edm::LogInfo("RPixDetClusterizer")<<" SeedVector size = "<<SeedVector_.size();
+
+// Looping on the seeds to make a cluster around the seed
+  for(std::vector<RPixCalibDigi>::iterator SeedIt = SeedVector_.begin(); SeedIt!=SeedVector_.end();++SeedIt){
+    make_cluster( *SeedIt, clusters);
+  }
+
+}
+
+
+void RPixDetClusterizer::make_cluster(RPixCalibDigi aSeed,  std::vector<CTPPSPixelCluster> &clusters ){
+
+/// check if seed already used
+  if(calib_rpix_digi_set_.size()==0 || calib_rpix_digi_set_.find(aSeed)==calib_rpix_digi_set_.end() ){
+    return;
+  }
+
+
+// creating a temporary cluster
+
+  tempCluster atempCluster;
+
+// filling the cluster with the seed
+  atempCluster.addPixel(aSeed.row(),aSeed.column(),aSeed.electrons());
+  calib_rpix_digi_set_.erase( aSeed );
+
+
+  while ( ! atempCluster.empty()) {
+  //This is the standard algorithm to find and add a pixel
+    auto curInd = atempCluster.top(); 
+    atempCluster.pop();
+    for ( auto c = std::max(0,int(atempCluster.col[curInd])-1); c < std::min(int(atempCluster.col[curInd])+2,maxCol) ; ++c) {
+      for ( auto r = std::max(0,int(atempCluster.row[curInd])-1); r < std::min(int(atempCluster.row[curInd])+2,maxRow); ++r)  {
+	
+	for(std::set<RPixCalibDigi>::iterator RPCDit2 = calib_rpix_digi_set_.begin(); RPCDit2 != calib_rpix_digi_set_.end(); ){
+	  if( (*RPCDit2).column() == c && (*RPCDit2).row() == r && (*RPCDit2).electrons() > ADCThreshold_*ElectronADCGain_ ){
+	    
+	    if(!atempCluster.addPixel( r,c,(*RPCDit2).electrons() )) {goto endClus;}
+	    RPCDit2 =  calib_rpix_digi_set_.erase(RPCDit2);
+	    
+	  }else{
+	    ++RPCDit2;
+	  }
+
+	}
+
+      }
+    }
+	     
+  }  // while accretion
+
+
+ endClus:
```

Comment:
The same as this: https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizer.cc#L459
I don't like it much but I took it as it is.